compression/compression.go

```diff
@@ -0,0 +1,263 @@
+package compression
+
+import (
+	"bufio"
+	"bytes"
+	"compress/bzip2"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+	"sync"
+
+	"github.com/containerd/log"
+	"github.com/klauspost/compress/zstd"
+)
+
+// Compression is the state represents if compressed or not.
+type Compression int
+
+const (
+	Uncompressed Compression = 0 // Uncompressed represents the uncompressed.
```

Comment:
This could be `None` instead of `Uncompressed` as suggested in https://github.com/moby/moby/pull/49167#discussion_r1895912656

I kept it to align with containerd, but TBD I guess?

Reply:
Not really a blocker for me, but I would be much more in favor of `None` (`compression.None` instead of `compression.Uncompressed`) 😅

Reply:
OK; done!

I also swapped the order of commits, doing the `test-table` changes first (before the move)